community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/ConcurrentMapState.java

```diff
@@ -73,19 +70,19 @@ public String toString()
     @Override
     public EntryUpdater<Key> updater( long version, Lock lock )
     {
-        if ( versionSequence.seen( version, META ) )
+        if ( version <= previousVersion )
         {
-            throw new IllegalStateException( "Cannot apply update with given version " + version +
-                                             " when base version is " + previousVersion );
+            return EntryUpdater.noUpdates();
         }
-        versionSequence.offer( version, META );
         update( highestAppliedVersion, version );
+        hasTrackedChanges.set( true );
```

Comment:
Just to check if my understanding is correct, is this necessary to make sure we return true from `hasChanges()` even before updating the `changes` map?